lib/acquia-http-hmac/rack_authenticate.rb

```diff
@@ -23,15 +23,20 @@ def call(env)
         return unauthorized if auth_header.empty?
 
         attributes = Acquia::HTTPHmac::Auth.parse_auth_header(auth_header)
+
+        # Pass the id to later stages
+        # Replaced id here to be able to log invalid requests too
+        env['ACQUIA-HTTP-HMAC-ID'] = attributes[:id]
```

Comment:
I believe that ACQUIA-HTTP-HMAC-ID is intended to indicate that the current execution has authenticated as that ID.  Moving it here will break code that makes this assumption.  A different environment variable could be used to store the attempted authentication ID prior to validating the authentication attempt.